cattle/plugins/host_info/disk.py

```diff
@@ -5,15 +5,40 @@
 
 
 class DiskCollector(object):
-    def __init__(self):
+    def __init__(self, docker_client=None):
         self.unit = 1048576
         self.cadvisor = CadvisorAPIClient(Config.cadvisor_ip(),
                                           Config.cadvisor_port())
 
+        self.docker_client = docker_client
+        self.docker_storage_driver = None
+
+        if self.docker_client:
+            self.docker_storage_driver = \
+                self.docker_client.info().get("Driver", None)
+
     def _convert_units(self, number):
         # Return in MB
         return round(float(number)/self.unit, 3)
 
+    def _get_dockerstorage_info(self):
+        data = {}
+
+        if self.docker_client:
+            for item in self.docker_client.info().get("DriverStatus"):
+                data[item[0]] = item[1]
```

Comment:
Discussed offline. There is higher level error handling that would catch these problems.